### ISSUE TITLE
Hummingbot v1.6. Strat: amm_arb. Error msg:  "mm_arb - uniswap_ethereum_mainnet not ready: waiting for allowances."

### ISSUE THREAD

**Issue by unclecharliecrypto** [hummingbot/hummingbot#5612]
Overview
In Hummingbot v1.6, when attempting to run the amm_arb strategy using connector_1 "uniswap_ethereum_mainnet" several errors appear - including the 1 above. These errors are circled in red in attached screenpic.

**Steps To Reproduce**
1. Environment: MacOS (M1). Hummingbot v1.6. Docker.
2. Start & check2 Docker containers (hummingbot-gateway-xxxx, hummingbot-instance-amm_arb) status "running".
3. In terminal, run %docker attach gb-hummingbot-instance-amm_arb_files (my naming syntax fyi) & Hummingbot GUI comes up.
4. Check Gateway: RUNNING, ONLINE in Hummingbot GUI
5. Check connections to 2 relevant exchanges. I do this by running balances and see both my ftx and uniswap balances.
6. Import amm_arb strategy (config in attached screen pic)
7. Start bot. Error msgs are circled in red in screen pic. 
8. Metamask wallet contains ETH, WETH & USDC. Metamask is connected to hummingbot.org.
<img width="1711" alt="Screen Shot 2022-08-23 at 21 21 09" src="https://user-images.githubusercontent.com/77430575/186156620-7af0d713-3031-4e19-80b8-7faf58ec57ba.png">
9. Carlito via Discord confirmed both selected pairs ok for amm_arb.
**Screenshots**
PFA

Discord User Name: CGB_Happy

**Comment by nikspz** [comment]
@unclecharliecrypto most likely related to your nodeURL for ethereum mainnet, you could try to use infura

**Comment by unclecharliecrypto** [comment]
@nikspz. Thanks nikspz. Let me check on infura & revert.

**Comment by unclecharliecrypto** [comment]
@nitkspz
Hello!   
Wondering how to (in Hummingbot v1.6 - mac0S  M1 - Docker installation) get Infura registered? 
I attach some earlier documentation (screen pic) showing how but for example >>connect ethereum command appears  no longer valid in Hummingbot v1.6
I have setup an Infura account.
For example how in Hummingbot v1.6
<img width="1401" alt="Screen Shot 2022-08-24 at 18 44 54" src="https://user-images.githubusercontent.com/77430575/186391436-5cecf5db-ffe3-43ec-bc8f-bf6912c5c4e2.png">
 GUI to change the default Ethereum mainnet address https://main-rpc.linkpool.io/. to that of infura. https://mainnet.infura.io....

Any help greatly appreciated!

**Comment by unclecharliecrypto** [comment]
@nikspz. Hello! I see 3 choices for infura in the ethereum.yml fil: ropsten, optimism & kovan.
I am not too familiar with these wallets - from a previous vid I thought 1 or more of them were test wallets? (similar to say a CEX bitmex_testnet). For real testing where I could link my metamask wallet to my infura account I wonder which of the 3 is the best?

**Comment by nikspz** [comment]
@unclecharliecrypto 
Sorry for inconvenience, we're currently updating nodeURL logic and on the way to new release

**Comment by nikspz** [comment]
@unclecharliecrypto you need to use gateway config ethereum ..... mainnet  ..... nodeURL or nodeAPIKey
let me check exactly

**Comment by unclecharliecrypto** [comment]
@nikspz Thank you so much!

**Comment by nikspz** [comment]
@unclecharliecrypto 
for kovan (testnet):
`gateway config ethereum.networks.kovan.nodeURL your_url_for_kovan`
then 
`gateway start` to restart gateway
![image](https://user-images.githubusercontent.com/83953535/186399887-2079f66d-60bf-4c68-894f-ac7aae71007f.png)
![image](https://user-images.githubusercontent.com/83953535/186400745-3dbb68da-695e-487f-a059-fcb4a6768d51.png)


for mainnet 
`gateway config ethereum.networks.mainnet.nodeURL your_url_for_mainnet`
then 
`gateway start` to restart gateway
here you could find it:
![image](https://user-images.githubusercontent.com/83953535/186400635-e80d9919-b207-46d1-8281-d5d7ac3f37bc.png)

**Comment by unclecharliecrypto** [comment]
@nikspz Really appreciate your help!

**Comment by unclecharliecrypto** [comment]
@nikspz Hello! Hey just fyi I was able to change over to the infura address which comes up as a prompt (screen pic attached - thank you!). Next question is  Getting error "The private key param is not at a valid Ethereum private key (64 hexadecimal characters). Now I have connected my Metamask wallet into my Infura account. Basically my q. is from which app (Metamask or Infura?)...Version 1.6 Hummingbot on Mac0S via Docker
<img width="366" alt="Screen Shot 2022-08-28 at 22 07 29" src="https://user-images.githubusercontent.com/77430575/187076191-bc6081c8-5463-407f-975e-cceb633c5726.png">

.
